### ISSUE TITLE
not-an-iterable error with attrs and type hinting.

### ISSUE THREAD

**Issue by tarcisioe** [PyCQA/pylint#2766]
### Steps to reproduce
1. Create a type-hinted class using attrs with an iterable member and try to iterate over an instance member.

```python
from typing import List

import attr


@attr.s
class A:
    i: List[str] = attr.ib()


for x in A().i:
    print(x)
```

2. Run Pylint on the code:

```
$ pylint test.py -d missing-docstring
```

### Current behavior
```                                                                                            
************* Module test
test.py:11:9: E1133: Non-iterable value A().i is used in an iterating context (not-an-iterable)
```

### Expected behavior
No such error, since the member is iterable.

### pylint --version output
```
$ pylint --versionpylint 2.2.2
astroid 2.1.0
Python 3.6.5 | packaged by conda-forge | (default, Apr  6 2018, 13:39:56)
[GCC 4.8.2 20140120 (Red Hat 4.8.2-15)]
```

Related to python-attrs/attrs#398

**Comment by PCManticore** [comment]
Thanks for reporting an issue @tarcisioe `pylint` doesn't have full support for `attrs` right now, which is why you are getting this error.

**Comment by bohush** [comment]
@PCManticore  And is there any plan to fix it / enhance the support for `attrs` in near feature?
Or do you have any suggestion for a workaround?
Thanks.

**Comment by PCManticore** [comment]
@bohush I don't have any plans for `attrs` for the near future. We're pretty limited in terms of manpower so we have to be selective in terms of what gets done. Usually for popular libraries that need custom support in `astroid`, our inference engine, we're letting contributors to fill the gaps in terms of PRs and contributions. We already have *some* support for `attrs` in `astroid` [here](https://github.com/PyCQA/astroid/blob/master/astroid/brain/brain_attrs.py) but it's definitely not enough to figure out all the magic that `attrs` does.

I'd also like to mention that various opinions expressed by "luminaries" from the linked issue with regard to what `pylint` can do are just that, opinions. Writing a static analysis tool that works across the board for the entire Python ecosystem is hard, especially when it's done by volunteers. Yes, maybe `mypy` is better for that, but it also has full time employees of Dropbox that drive those efforts.

**Comment by collinstocks-oscar** [comment]
Do you know how to work around the problem? Is there a way to force pylint to treat a variable as a particular type? Or do you have to disable the error everywhere where it occurs?

**Comment by hippo91** [comment]
@collinstocks-oscar i'm not aware of a specific work around for this problem. Sorry.

**Comment by collinstocks-oscar** [comment]
Thanks. I'm sure this isn't the right place for a feature request, but it would be excellent if inferred types could be overridden with an annotation. Sometimes I just know better than Pylint.

**Comment by hippo91** [comment]
@collinstocks-oscar feature request are always welcomed! For now `pylint` doesn't take into account annotation but it would be, of course, a great improvment. Feel free to make a PR to begin this work if you want to!

**Comment by Pierre-Sassoulas** [comment]
This has been fixed in 2.14, or I cannot reproduce it.

**Comment by xqgex** [comment]
With pylint `2.17.0` and attrs `22.1.0`:

Code:

```Python
""" Module description. """

from typing import List

from attrs import define, field


@define
class Foo:  # pylint: disable=too-few-public-methods
    """ Class description. """
    var: List[int] = field()


for x in Foo(var=[1,2,3]).var:
    print(x)
```

Command: `python3 -m pylint test.py`

Output:

```
************* Module test
test.py:14:9: E1133: Non-iterable value Foo(var=[1, 2, 3]).var is used in an iterating context (not-an-iterable)

------------------------------------------------------------------
Your code has been rated at 1.67/10
```
